tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Graph.java

```diff
@@ -1091,6 +1059,28 @@ private static GraphDef toGraphDef(TF_Graph handle) {
     }
   }
 
+  private GraphDef addOrUpdateInit(GraphDef graphDef) {
+    if (newInitializersMarker < 0) {
+      return graphDef;
+    }
+    var graphDefWithInitBuilder = graphDef.toBuilder();
+    var initNode =
+        graphDefWithInitBuilder.getNodeBuilderList().stream()
+            .filter(n -> n.getName().equals(INIT_OP_NAME))
+            .findFirst()
+            .orElseGet(
+                () -> {
+                  return graphDefWithInitBuilder
+                      .addNodeBuilder()
+                      .setName(INIT_OP_NAME)
+                      .setOp(NoOp.OP_NAME);
+                });
+    initializers.stream()
+        .skip(newInitializersMarker)
+        .forEach(op -> initNode.addInput("^" + op.name()));
```

Comment:
What's the `^` mean in this case?

Reply:
This is how you mark an input as a control dependency in the proto file. But I agree that is worth commenting it out :)

Reply:
Sounds good, and yes it definitely needs a comment.